autogluon/utils/tabular/features/generators/category.py

```diff
@@ -0,0 +1,121 @@
+            'alphanumeric' : Sort the categories alphanumerically.
+            'count' : Sort the categories by frequency (Least frequent in front with code of 0)
+    minimum_cat_count : int, default None
+        The minimum number of occurrences a category must have in the training data to avoid being considered a rare category.
+        Rare categories are removed and treated as missing values.
+        If None, no minimum count is required. This includes categories that never occur in the data but are present in the category object as possible categories.
+    maximum_num_cat : int, default None
+        The maximum amount of categories that can be considered non-rare.
+        Sorted by occurrence count, up to the N highest count categories will be kept if maximum_num_cat=N. All others will be considered rare categories.
+    **kwargs :
+        Refer to AbstractFeatureGenerator documentation for details on valid key word arguments.
+    """
+    def __init__(self, stateful_categories=True, minimize_memory=True, cat_order='original', minimum_cat_count: int = None, maximum_num_cat: int = None, **kwargs):
+        super().__init__(**kwargs)
+        self._stateful_categories = stateful_categories
+        if minimum_cat_count is not None and minimum_cat_count <= 1:
+            minimum_cat_count = None
+        if cat_order not in ['original', 'alphanumeric', 'count']:
+            raise ValueError(f"cat_order must be one of {['original', 'alphanumeric', 'count']}, but was: {cat_order}")
+        self.cat_order = cat_order
+        self._minimum_cat_count = minimum_cat_count
+        self._maximum_num_cat = maximum_num_cat
+        self.category_map = None
+
+        if minimize_memory:
+            self._post_generators = [CategoryMemoryMinimizeFeatureGenerator(inplace=True)] + self._post_generators
+
+    def _fit_transform(self, X: DataFrame, **kwargs) -> (DataFrame, dict):
+        if self._stateful_categories:
+            X_out, self.category_map = self._generate_category_map(X=X)
+        else:
+            X_out = self._transform(X)
+        feature_metadata_out_type_group_map_special = copy.deepcopy(self.feature_metadata_in.type_group_map_special)
+        if S_TEXT in feature_metadata_out_type_group_map_special:
+            text_features = feature_metadata_out_type_group_map_special.pop(S_TEXT)
+            feature_metadata_out_type_group_map_special[S_TEXT_AS_CATEGORY] += [feature for feature in text_features if feature not in feature_metadata_out_type_group_map_special[S_TEXT_AS_CATEGORY]]
+        return X_out, feature_metadata_out_type_group_map_special
+
+    def _transform(self, X: DataFrame) -> DataFrame:
+        return self._generate_features_category(X)
+
+    def _infer_features_in(self, X: DataFrame, y: Series = None) -> list:
+        return self.feature_metadata_in.get_features(valid_raw_types=[R_OBJECT, R_CATEGORY], invalid_special_types=[S_DATETIME_AS_OBJECT])
```

Comment:
I agree and had a TODO for this in pipeline.py. I moved the TODO from pipeline.py to abstract.py for increased visibility. The exploration of this is certainly worthwhile, but will not be part of this PR to avoid changing prior functionality.